napalm/base/base.py

```diff
@@ -91,8 +91,8 @@ def _netmiko_open(self, device_type, netmiko_optional_args=None):
                 timeout=self.timeout,
                 **netmiko_optional_args
             )
-        except NetMikoTimeoutException:
-            raise ConnectionException("Cannot connect to {}".format(self.hostname))
+        except NetMikoTimeoutException as e:
+            raise ConnectionException("Cannot connect to {} - {}".format(self.hostname, e))
```

Comment:
Should probably use newlines between the NAPALM message and the message contained in "e" i.e. something like:

```python
raise ConnectionExcpetion(f"Cannot connect to {self.hostname}\n\n{e}")
```

So it becomes less confusing with the two strings being combined together.

Reply:
It's somewhat unconventional to use multiline error messages at all in python exceptions.
E.g. don't think stdlib does it anywhere I can remember, and quick google suggests that people can even be surprised by it and don't expect this ( https://github.com/filp/whoops/issues/298 https://stackoverflow.com/questions/51449888/ ... ).

But on the other hand, this specific error message is also known to be multiline in advance, so guess adding more newlines here shouldn't really be a problem.
Updated the PR to put two newlines instead of a dash.

Although without newline, IMO error message will work nicely as a more common `one-line summary<newline><newline>details` format, used in all git commits and many other places, as a middle-ground between single-line and multi-line formats.
I.e. something like this:
```
ConnectionException: Cannot connect to 10.0.0.98 - TCP connection to device failed.

Common causes of this problem are:
1. Incorrect hostname or IP address.
2. Wrong TCP port.
3. Intermediate firewall blocking access.

Device settings: cisco_ios 10.0.0.98:22
```

But again, don't think it matters too much, as long as underlying error is actually reported in some way.